### DISCUSSION TITLE
Serverless functions fail as "too big" when using Vercel for Github auto-deploy

### DISCUSSION THREAD

**Discussion by bigsee** [vercel/vercel#4709] [Help] [answered]
I built a small app - a static website with a single serverless function that provides an integration for newsletter signup via the Mailchimp API. The serverless function is written in Python.

When I deploy this site using `vercel` and then `vercel alias staging.mydomain.com`, everything is deployed fine.

When I auto-deploy exactly the same code to the same url using the Github integration, it fails to build telling me that the serverless function is too big (85MB). The function code itself is about 1KB, with one module in requirements.txt.

I have the Github integration set up in Vercel so that pushing to branch `staging` deploys to `staging.mydomain.com`, as above.

The discussion here ([#4114](https://github.com/vercel/vercel/discussions/4114)) states that the size of the serverless function takes into account static files. However, my dist/ folder is only 5MB. This is the first time I've tried playing with the Vercel for Github integration so I might be doing something wrong.

Anyone know what I've missed / misunderstood? Or is this a bug?

**Comment by styfle** [comment]
Hi @bigsee 

What is in `requirements.txt`? Maybe the dependency is large?

Also, what other files do you have in the repo? Is your project open source?

**Comment by bigsee** [comment]
Hi @styfle 

Thank you for the response.

Requirements.txt only contains one module - requests.

Are the assets for the static site counted in the serverless function size? The assets/ folder is quite big.

I'm afraid it's not open-source but I'd be happy to add you as a contributor if you'd like to take a look. As I say, the workaround of building, deploying and aliasing manually works perfectly fine. Just seems odd that it works that way but not with the GitHub auto-deployment.

**Comment by styfle** [comment, marked as answer]
> Are the assets for the static site counted in the serverless function size?

Yes we don't have static analysis for python functions so all the files are included with the function.

You can add the `excludesFiles` property in `vercel.json` to ignore static assets.

For example, exclude the `public` directory for all your python functions:

```json
{
  "functions": {
    "api/**/*.py": {
      "excludeFiles": "public/**"
    }
  }
}
```

Related to #2830

**Comment by bigsee** [comment]
@styfle OK, all clear - thanks! I was trying the zero-config approach. Since you linked an open issue, and since I didn't have this issue on another project that used node functions inside the api/ folder, I'll assume that this isn't intentional behaviour.

I will use your helpful workaround for now. 👍🏽
